src/stream/src/task/stream_manager.rs

```diff
@@ -163,6 +163,9 @@ impl LocalStreamManager {
         await_tree_config: Option<await_tree::Config>,
         watermark_epoch: AtomicU64Ref,
     ) -> Self {
+        // set the global `CONFIG` variable
+        _ = crate::CONFIG.set(env.config().clone());
```

Comment:
Under `madsim` the compute node (then `LocalStreamManager`) can be started and killed for many times. Will the global variable become problematic?

Reply:
Personally I'll +1 for task-local approaches as it does not introduce any races. 🤣

Reply:
> Under `madsim` the compute node (then `LocalStreamManager`) can be started and killed for many times. Will the global variable become problematic?

If the streaming config can change during the time being killed, maybe the global variable approach is problematic.